app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt

```diff
@@ -34,23 +37,28 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.kks.nimblesurveyjetpackcompose.R
+import com.kks.nimblesurveyjetpackcompose.ui.presentation.common.ErrorAlertDialog
+import com.kks.nimblesurveyjetpackcompose.ui.presentation.common.Loading
+import com.kks.nimblesurveyjetpackcompose.ui.presentation.destinations.HomeScreenDestination
 import com.kks.nimblesurveyjetpackcompose.ui.theme.Concord
 import com.kks.nimblesurveyjetpackcompose.ui.theme.White18
 import com.kks.nimblesurveyjetpackcompose.util.extensions.loginTextFieldModifier
 import com.kks.nimblesurveyjetpackcompose.viewmodel.splash.SplashViewModel
 import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.ramcosta.composedestinations.navigation.EmptyDestinationsNavigator
 import kotlinx.coroutines.launch
 
 @Destination(start = true)
 @Composable
-fun SplashScreen(viewModel: SplashViewModel = hiltViewModel()) {
+fun SplashScreen(navigator: DestinationsNavigator, viewModel: SplashViewModel = hiltViewModel()) {
     val showLoginComponents by viewModel.shouldNavigateToLogin.collectAsState()
+    val shouldShowLoading by viewModel.shouldShowLoading.collectAsState()
+    val shouldShowError by viewModel.hasError.collectAsState()
+    val isLoginSuccess by viewModel.isLoginSuccess.collectAsState()
     val logoOffset = remember { Animatable(Offset(0f, 0f), Offset.VectorConverter) }
     val positionToAnimate = -LocalDensity.current.run { 221.dp.toPx() }
-
-    LaunchedEffect(
-        key1 = showLoginComponents
-    ) {
+    LaunchedEffect(keys = arrayOf(showLoginComponents)) {
```

Comment:
Perhaps a newline here for readability? 😄 
```suggestion

    LaunchedEffect(keys = arrayOf(showLoginComponents)) {
```